### PR TITLE
Feat/hex support

### PR DESCRIPTION
**Pull request by fiveisgreen**
Hexidecimal register support added, as well as a test for Hex support.

### CODE REVIEW

File: src/CommandBasedBackendRegisterAccessor.cpp
Comment (killenb) [inline src/CommandBasedBackendRegisterAccessor.cpp:1]: Line 206 does not produce hex representation strings.

File: tests/DummyServerLib/DummyServer.cpp
Comment (killenb) [inline tests/DummyServerLib/DummyServer.cpp:201]: You picked an unnecessarily complicated syntax for the hex command. Just positional, fixed number arguments would be sufficient. The rest is tested already, so we can concentrate on the hex part of it.
We should also test the mixed writing (last argument must not have 0x extension)
```HEX 0x12bab 0xd0d010 acbdef```
```
HEX?
0x12bab 0xd0d010 acbdef
```

File: tests/testCommandBasedBackendUnified.cpp
Comment (killenb) [inline tests/testCommandBasedBackendUnified.cpp:261]: Hm, the test is writing. It is enabled, but why does it not fail?

File: tests/testCommandBasedBackendUnified.cpp
Comment (killenb) [inline tests/testCommandBasedBackendUnified.cpp:281]: Aha, that's why the test passes. It does not require the text to be a hex representation. This is just taking the integer string, interprets it as integer and returns the correct number.
 It is missing force0xHex() here.
 
 I am wondering if we should interpret the string and store the value, and then getRemoteValue() returns the stored value. In principle it does not matter where we interpret the string. However, we also need it in setRemoteValue, so we should do it in  DummyServer::setHex().

**Review by killenb — CHANGES_REQUESTED** [review]
The writing is not working. The test does not detect it due to a bug.

What is also needed it to test that we can define writing with and without the leading ```0x```.

File: src/CommandBasedBackendRegisterAccessor.cpp
Comment (fiveisgreen) [reply src/CommandBasedBackendRegisterAccessor.cpp:182]: While this is nice, it creates a hazard where 0x is omitted in the config file, resulting in userTypeToUserType receiving "0x0x1234". So, there ought to then be a check that that 0x is absent, else throw an error.

File: include/stringUtils.h
Comment (fiveisgreen) [reply include/stringUtils.h:58]: Ok, we can do that. We still need that logic in the DummyServer, but can be moved there. Removing it also creates the issue

File: src/CommandBasedBackend.cpp
Comment (fiveisgreen) [reply src/CommandBasedBackend.cpp:1]: The correction to the comment needs a home. Should it go directly into the master, or some new patch branch?

**Review by killenb — APPROVED** [review]
Approved after applying the suggestions.

Before merging:
* Test that all tests are passing (coding style test was again failing!)
* Squash everything into one commit (onto 6989ba9)
* Rebase onto master

**Pull request merged**
